cpukit/score/src/schedulerstrongapa.c

```diff
@@ -3,339 +3,820 @@
+        curr_priority = SCHEDULER_PRIORITY_PURIFY( curr_priority );
+
+        if ( curr_priority > max_priority_num ) {
+          lowest_scheduled = curr_node;
+          max_priority_num = curr_priority;
+        }
+      }
+    }
+  }
+
+  _Assert( lowest_scheduled != NULL );
+  return lowest_scheduled;
 }
 
-static void _Scheduler_strong_APA_Do_update(
+static inline void _Scheduler_strong_APA_Extract_from_scheduled(
   Scheduler_Context *context,
-  Scheduler_Node *node_to_update,
-  Priority_Control new_priority
+  Scheduler_Node    *node_to_extract
 )
 {
-  Scheduler_strong_APA_Context *self =
-    _Scheduler_strong_APA_Get_self( context );
-  Scheduler_strong_APA_Node *node =
-    _Scheduler_strong_APA_Node_downcast( node_to_update );
-
-  _Scheduler_SMP_Node_update_priority( &node->Base, new_priority );
-  _Scheduler_priority_Ready_queue_update(
-    &node->Ready_queue,
-    SCHEDULER_PRIORITY_UNMAP( new_priority ),
-    &self->Bit_map,
-    &self->Ready[ 0 ]
-  );
+  Scheduler_strong_APA_Context *self;
+  Scheduler_strong_APA_Node    *node;
+
+  self = _Scheduler_strong_APA_Get_self( context );
+  node = _Scheduler_strong_APA_Node_downcast( node_to_extract );
+
+  _Scheduler_SMP_Extract_from_scheduled( &self->Base.Base, &node->Base.Base );
+  /* Not removing it from Ready since the node could go in the READY state */
 }
 
-static Scheduler_strong_APA_Context *
-_Scheduler_strong_APA_Get_context( const Scheduler_Control *scheduler )
+static inline void _Scheduler_strong_APA_Extract_from_ready(
+  Scheduler_Context *context,
+  Scheduler_Node    *node_to_extract
+)
 {
-  return (Scheduler_strong_APA_Context *) _Scheduler_Get_context( scheduler );
+  Scheduler_strong_APA_Context *self;
+  Scheduler_strong_APA_Node    *node;
+
+  self = _Scheduler_strong_APA_Get_self( context );
+  node = _Scheduler_strong_APA_Node_downcast( node_to_extract );
+
+  if( !_Chain_Is_empty( &self->Ready ) ) {
+    if( !_Chain_Is_node_off_chain( &node->Ready_node ) ) {
+      _Chain_Extract_unprotected( &node->Ready_node );
+      _Chain_Set_off_chain( &node->Ready_node );
+    }
+  }
 }
 
-void _Scheduler_strong_APA_Initialize( const Scheduler_Control *scheduler )
+static inline void _Scheduler_strong_APA_Insert_ready(
+  Scheduler_Context *context,
+  Scheduler_Node    *node_base,
+  Priority_Control   insert_priority
+)
 {
-  Scheduler_strong_APA_Context *self =
-    _Scheduler_strong_APA_Get_context( scheduler );
+  Scheduler_strong_APA_Context *self;
+  Scheduler_strong_APA_Node    *node;
 
-  _Scheduler_SMP_Initialize( &self->Base );
-  _Priority_bit_map_Initialize( &self->Bit_map );
-  _Scheduler_priority_Ready_queue_initialize(
-    &self->Ready[ 0 ],
-    scheduler->maximum_priority
-  );
+  self = _Scheduler_strong_APA_Get_self( context );
+  node = _Scheduler_strong_APA_Node_downcast( node_base );
+
+  if( _Chain_Is_node_off_chain( &node->Ready_node ) ) {
+    _Chain_Append_unprotected( &self->Ready, &node->Ready_node );
+  }
+  else {
```

Comment:
combine in one line `} else {`

Reply:
And the indents following this else remains the same, right?

Reply:
Thank you.